Sources/OTelTesting/OTelInMemoryLogRecordProcessor.swift

```diff
@@ -15,30 +15,30 @@ import NIOConcurrencyHelpers
 @_spi(Logging) import OTel
 
 /// An in-memory log record processor, collecting emitted log records into ``onEmit(_:)``.
-package final class OTelInMemoryLogRecordProcessor: OTelLogRecordProcessor {
-    package var records: [OTelLogRecord] { _records.withLockedValue { $0 } }
-
-    private let _records = NIOLockedValueBox([OTelLogRecord]())
-    private let _numberOfShutdowns = NIOLockedValueBox(0)
-    private let _numberOfForceFlushes = NIOLockedValueBox(0)
+@_spi(Logging)
+public actor OTelInMemoryLogRecordProcessor: OTelLogRecordProcessor {
```

Comment:
Do we want to make this a `class` instead of an `actor` given the properties are explicitly marked as `nonisolated`?

Reply:
@czechboy0 and @simonjbeaumont, what do you think?

Reply:
In the interest of getting this PR merged I went ahead and made the change. Happy to revisit later if we want to go with an alternative approach.